include/drivers/pcie/pcie.h

```diff
@@ -169,8 +179,17 @@ extern void pcie_irq_enable(pcie_bdf_t bdf, unsigned int irq);
 
 #define PCIE_CONF_BAR_IO(w)		(((w) & 0x00000001U) == 0x00000001U)
 #define PCIE_CONF_BAR_MEM(w)		(((w) & 0x00000001U) != 0x00000001U)
-#define PCIE_CONF_BAR_64(w)		(((w) & 0x00000006U) == 0x00000004U)
-#define PCIE_CONF_BAR_ADDR(w)		((w) & ~0xfUL)
+#define PCIE_CONF_BAR_64(w)		(((w) & 0x00000007U) == 0x00000004U)
+#define PCIE_CONF_BAR_IO_ADDR(w)	((w) & ~0x3U)
+#define PCIE_CONF_BAR_MEM_ADDR(w)	((w) & ~0xfUL)
+
+static inline uintptr_t pcie_conf_bar_addr(uintptr_t w)
+{
+	return PCIE_CONF_BAR_MEM(w) ? PCIE_CONF_BAR_MEM_ADDR(w)
+				: PCIE_CONF_BAR_IO_ADDR(w);
+}
+#define PCIE_CONF_BAR_ADDR		pcie_conf_bar_addr
```

Comment:
Since you are skipping I/O ones, there is no need for these anymore, and can just use the old `PCIE_CONF_BAR_ADDR()`.